server/src/main/scala/services/PcpArrival.scala

```diff
@@ -87,11 +87,23 @@ object PcpArrival {
 
   def gateOrStandWalkTimeCalculator(gateWalkTimesProvider: GateOrStandWalkTime,
                                     standWalkTimesProvider: GateOrStandWalkTime,
-                                    defaultWalkTimeMillis: MillisSinceEpoch)(flight: Arrival): MillisSinceEpoch = {
-    val walkTime = standWalkTimesProvider(flight.Stand.getOrElse(""), flight.Terminal).getOrElse(
-      gateWalkTimesProvider(flight.Gate.getOrElse(""), flight.Terminal).getOrElse(defaultWalkTimeMillis))
-    log.debug(s"walkTimeForFlight ${Arrival.summaryString(flight)} is $walkTime millis ${walkTime / 60000} mins default is $defaultWalkTimeMillis")
-    walkTime
+                                    defaultWalkTimeMillis: MillisSinceEpoch,
+                                    coachTransfer: List[CoachTransfer])(flight: Arrival): MillisSinceEpoch = {
+    val greenListTerminal = List(T2, T3, T5)
```

Comment:
I think we need to encapsulate this logic in a service that is specific to the airport. It's not just LHR that has T2, T3 & T5

We also need to not apply this logic if we're viewing a date before T4 opened for direct red list flights

Reply:
I thought about it but was thinking it could a lot of config to add coach approach and change once we have to appy to other ports . But that you have comment . I can move to config .